plugins/arc/src/ArcRenderer/ArcRendering.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { readConfObject } from '@jbrowse/core/configuration'
+import { bpSpanPx, measureText } from '@jbrowse/core/util'
+import { observer } from 'mobx-react'
+import { Tooltip } from 'react-svg-tooltip'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function ArcRendering(props: any) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onClick = (event: any, id: any) => {
+    const { onFeatureClick: handler } = props
+    if (!handler) {
+      return undefined
+    }
+    return handler(event, id)
+  }
+
+  const {
+    features,
+    config,
+    regions,
+    blockKey,
+    bpPerPx,
+    displayModel: { selectedFeatureId },
+  } = props
+  const [region] = regions
+  const arcsRendered = []
+
+  for (const feature of features.values()) {
+    const [left, right] = bpSpanPx(
+      feature.get('start'),
+      feature.get('end'),
+      region,
+      bpPerPx,
+    )
+
+    const featureId = feature.id()
+    const id = blockKey + '-' + featureId
+    let stroke = readConfObject(config, 'color', { feature })
+    let textStroke = 'black'
+    if (
+      selectedFeatureId &&
+      String(selectedFeatureId) === String(feature.id())
+    ) {
+      stroke = textStroke = 'red'
+    }
+    const label = readConfObject(config, 'label', { feature })
+    const caption = readConfObject(config, 'caption', { feature })
+    const strokeWidth = readConfObject(config, 'thickness', { feature })
```

Comment:
might want to use `readConfObject(config, 'thickness', { feature })||1` in case thickness is undefined (often on case where gene turns into arc track track it is since gene score might be undefined)